src/main/java/appeng/container/implementations/ContainerCraftAmount.java

```diff
@@ -76,5 +76,8 @@ public void setItemToCraft(@Nonnull final IAEItemStack itemToCreate) {
 
     public void openConfirmationGUI(EntityPlayer player, TileEntity te) {
         Platform.openGUI(player, te, this.getOpenContext().getSide(), GuiBridge.GUI_CRAFTING_CONFIRM);
+        if (player.openContainer instanceof ContainerCraftConfirm ccc) {
+            ccc.setItemToCraft(this.itemToCreate);
+        }
```

Comment:
It appears this is invoked from the set amount window, so another method (such as ctrl-clicking the NEI question mark) which bypasses this window while a craft is occurring likely results in it checking to merge with an empty ByteBuf, leading to the crash described.

Reply:
Huge thanks for narrowing this down for me, this was exactly it. Didn't think that this would be called from other mods but now I know!